### PR TITLE
Alfresco 6.0 support

### PR DESCRIPTION
**Pull request by kerkhofsd**
## Description
This PR adds build artifacts for and integration tests against Alfresco 6.0

## How Has This Been Tested?
Both 5 and 6 artifacts have been manually tested. Include integration tests for Dynamic Extensions on Alfresco 6.0 community.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the changelog.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
-  - oraclejdk8
+- oraclejdk8
+
+services:
+- docker
 
 script:
-  - ./gradlew build test
+- ./gradlew check
+- ./gradlew integrationTest -i --stacktrace
```
Comment (vierbergenlars) [inline .travis.yml]: You can add `integrationTest` to the `check` task

File: integration-tests/build.gradle [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        testClassesDirs = project.parent.sourceSets.test.output.classesDirs
+        classpath = project.parent.sourceSets.test.runtimeClasspath
+        outputs.upToDateWhen { false }
+
+        testLogging {
+            showStandardStreams = true
+            exceptionFormat = 'full'
+        }
+    }
+
```
Comment (vierbergenlars) [inline integration-tests/build.gradle]:
```suggestion

check.dependsOn integrationTest
```

File: integration-tests/build.gradle [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        useComposeFiles = [project.parent.projectDir.toString() + '/src/test/compose/docker-compose.yml']
+
+        removeVolumes = true
+        captureContainersOutput = true
+
+        // Uncomment for quick iterations when developing integration tests
+//        stopContainers = false
+
+        // expose alfresco/inflow on a random port, comment to use default port (8080)
+        environment.put 'COMPOSE_ALFRESCO_TCP_8080', '8080'
```
Comment (vierbergenlars) [inline integration-tests/build.gradle]: Do not start your environment variables with `COMPOSE_` or `DOCKER_`, as these prefixes are reserved for configuring `docker-compose` or the `docker` client.

File: integration-tests/src/test/compose/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                -Ddb.url=jdbc:postgresql://postgres:5432/alfresco
+                -Dindex.subsystem.name=noindex
+                -Djodconverter.enabled=false
+                -Ddeployment.method=DOCKER_COMPOSE
+                -Dcsrf.filter.enabled=false
+                -Xms1g -Xmx1g
+                -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8000
+                "
+    ports:
+    - ${COMPOSE_ALFRESCO_TCP_8080:-8080:8080} # by default expose on :8080
```
Comment (vierbergenlars) [inline integration-tests/src/test/compose/docker-compose.yml]: Shouldn't this be prefixed with `$DOCKER_IP`, so you can choose to bind to localhost or a specific IP?
```suggestion
    - $DOCKER_IP:${COMPOSE_ALFRESCO_TCP_8080:-8080:8080} # by default expose on :8080
```

File: integration-tests/src/test/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+ARG ALFRESCO_BASE_IMAGE
```
Comment (vierbergenlars) [inline integration-tests/src/test/docker/Dockerfile]: This file is not used, because you are using the `docker-alfresco` gradle plugin

**Comment by cetra3** [comment]
What's the status of this PR?

**Comment by tgeens** [comment]
The PR is functional, but it is a breaking change. As in: you should use DE 1.x for Alfresco 5.2 and below and DE 2.x (this PR) for Alfresco 6.x - so if you are creating extensions, you'd be required to create multiple builds with different dependencies.

We want to have another try at avoiding that, because it would push the cost down into the extensions.

It'll be another few weeks until we have time to work on this, but we all agree it's urgent.

**Comment by cetra3** [comment]
Yeah no worries, is there a premade amp?

I think that we might be stuck with two versions, as Alfresco 6 has a lot of core dependency changes.

**Comment by kerkhofsd** [comment]
@cetra3 
Building this PR results in the following AMP for Alfresco 6.0:
[alfresco-dynamic-extensions-repo-60-1.8.0-SNAPSHOT.amp.zip](https://github.com/xenit-eu/dynamic-extensions-for-alfresco/files/2944825/alfresco-dynamic-extensions-repo-60-1.8.0-SNAPSHOT.amp.zip)

Disclaimer: **this artifact still needs thorough testing and is not releasable AS-IS**. Just providing it as a sneak peek.

**Pull request merged**